### PR TITLE
TeamCity: add mac/arm64 build

### PR DESCRIPTION
**Pull request by nd**

### CODE REVIEW

**Comment by nd** [comment]
We finally added AppleM1 build agent.

**Comment by nd** [comment]
@aarzilli It looks like automatic build triggering doesn't work because of the created VCS root copy. It looks like the old root is used in pull requests feature settings. I think it can be fixed by detaching the VCS root from Aggregator and enabling 'show changes from dependencies' option instead.

**Comment by derekparker** [comment]
@nd that's my bad, I went around messing things up. I accidentally duplicated the VCS root in trying to update the user that is used for authentication. I created a new user in GH (DelveBot) and I want to use that for auth for TeamCity, but I think in doing so I've messed something up. I essentially want TeamCity to stop authing through my account, as it ends up using my profile picture in the aggregator response.

**Comment by nd** [comment]
Switched to DelveBot for GH operations. Hope it has all necessary permissions.

**Comment by derekparker** [comment]
@nd I've created an access token for DelveBot but I've forgotten where to input it in the TeamCity UI.

**Comment by aarzilli** [comment]
It seems that now linux/amd64 agents aren't starting?

**Comment by aarzilli** [comment]
PS. in any case the M1 builder is great news!

**Comment by nd** [comment]
> I've created an access token for DelveBot but I've forgotten where to input it in the TeamCity UI.

It should be specified [here](https://delve.beta.teamcity.com/admin/editProject.html?projectId=Delve&tab=versionedSettings&subTab=tokens). Generate Token and paste the GH token, after that created credentialsJSON:xxx can be used in settings.kts.

**Comment by derekparker** [comment]
@nd `credentialsJSON:a7e7526c-7195-4790-bbb6-9fb4692f92d0` should work then.

**Comment by nd** [comment]
updated

**Comment by nd** [comment]
> It seems that now linux/amd64 agents aren't starting?

not sure what caused it, but now they should start

**Comment by derekparker** [comment]
@nd looks like it still not getting triggered (or at least not updating the status on this PR). Is there still more config to fix or update?

**Comment by nd** [comment]
> looks like it still not getting triggered

TeamCity still uses old settings. Merging this PR into master should update them.

File: .teamcity/settings.kts [hunk 1]
```diff
@@ -49,7 +49,9 @@ val targets = arrayOf(
         "windows/amd64/tip",
 
         "mac/amd64/1.16",
-        "mac/amd64/tip"
+        "mac/amd64/tip",
+
+        "mac/arm64/1.16"
```
Comment (derekparker) [inline .teamcity/settings.kts:54]: Can you please also add tip for the arm builder.

Comment (nd) [reply .teamcity/settings.kts]: done

**Review by aarzilli — APPROVED** [review]
LGTM

**Comment by nd** [comment]
It looks like TC fails to publish status for PRs. @derekparker does DelveBot have repo:status and public_repo permissions?

**Comment by nd** [comment]
You can open 'Commit status publisher' and test connection [here](https://delve.beta.teamcity.com/admin/editBuildFeatures.html?id=buildType:Delve_AggregatorBuild).

**Comment by derekparker** [comment]
@nd thanks, updated.

**Comment by aarzilli** [comment]
@nd do you know if it's possible to have a remote shell on the M1, I'd like to investigate #2346 but it's awkward without it.

**Comment by nd** [comment]
> remote shell on the M1

Currently it is accessible from internal network only. Will try to organize an access from outside.

**Pull request merged**